src/manager/file.vala

```diff
@@ -0,0 +1,71 @@
+namespace ProtonPlus.Manager {
+    public class File {
+        public static void Extract (string install_location, string launcher_name, string tool_name) {
+            Archive.Read archive = new Archive.Read ();
+            archive.support_format_all ();
+            archive.support_filter_all ();
+
+            int flags;
+            flags = Archive.ExtractFlags.ACL;
+            flags |= Archive.ExtractFlags.PERM;
+            flags |= Archive.ExtractFlags.TIME;
+            flags |= Archive.ExtractFlags.FFLAGS;
+
+            Archive.WriteDisk ext = new Archive.WriteDisk ();
+            ext.set_standard_lookup ();
+            ext.set_options (flags);
+
+            if (archive.open_filename (install_location + tool_name + ".tar.gz", 1920000) != Archive.Result.OK) return;
+
+            ssize_t r;
+
+            unowned Archive.Entry entry;
+
+            int bob = 0;
+            string test = "";
+
+            for ( ;; ) {
+                r = archive.next_header (out entry);
+                if (r == Archive.Result.EOF) break;
+                if (r < Archive.Result.WARN) break;
+                entry.set_pathname (install_location + entry.pathname ());
+                if(bob++ == 0){
+                    test = entry.pathname ();
+                }
+                r = ext.write_header (entry);
```

Comment:
Why don't you check the content of `r` here? This `r` value is never used.

Reply:
r c'est juste pour stocker le resultat des operations. Elle est utiliser si tu regardes bien dans les if pour verifier il y a eu un warning ou si la fin du fichier (EOF) a ete atteint.

Reply:
Ouai, mais ce que tu assignes à la ligne 35 n'est jamais utilisé. Ajoute un if pour checker la valeur ou enlève l'assignation?

Reply:
Ah ok, j'avais enlever des messages d'erreurs, mais je les ai remis. La, la valeur est utiliser par apres. Merci :)